system/modules/core/library/Contao/Environment.php

```diff
@@ -311,6 +311,11 @@ protected static function httpXForwardedHost()
 	 */
 	protected static function ssl()
 	{
+		if (isset($_SERVER['HTTP_X_FORWARDED_PROTO']) && in_array($_SERVER['REMOTE_ADDR'], trimsplit(',', \Config::get('proxyServerIps'))))
+		{
+		    return $_SERVER['HTTP_X_FORWARDED_PROTO'] == 'https';
```

Comment:
I still don't think this makes sense, because the remote IP being a proxy does not tell us the header is originating from the proxy. As stated in the issue (if I remember correctly), there's no way to know. The best would be for that specific case to use the `initconfig.php`. Anyway, if we can agree this is not unsafe, I'm fine with keeping it.